pyro/distributions/torch.py

```diff
@@ -68,7 +50,15 @@ def sample(self, sample_shape=torch.Size()):
                     shift = (mean - variance).round()
                     result = torch.poisson(variance.expand(shape))
                     result = torch.min(result + shift, self.total_count)
-                    return torch.where(p < q, result, self.total_count - result)
+                    sample = torch.where(p < q, result, self.total_count - result)
+                # Draw exact samples for remaining items.
+                if exact.any():
+                    total_count = torch.where(exact, self.total_count,
+                                              torch.zeros_like(self.total_count))
+                    exact_sample = torch.distributions.Binomial(
+                        total_count, self.probs, validate_args=False).sample(sample_shape)
+                    sample = torch.where(exact, exact_sample, sample)
```

Comment:
i'm confused. how does L60 work if L53 didn't execute?

Reply:
that can't happen; the indentation on L53 is for the `with torch.no_grad()`, not an if statement. L60 can only execute if L53 executes.